populus/project.py

```diff
@@ -140,12 +149,21 @@ def get_source_file_hash(self):
             in source_file_paths
         )).hexdigest()
 
+    def get_source_modification_time(self):
+        source_file_paths = find_project_contracts(self.project_dir)
+        return max(
+            os.path.getmtime(source_file_path)
+            for source_file_path
+            in source_file_paths
+        ) if len(source_file_paths) > 0 else None
+
     @property
     def compiled_contracts(self):
-        source_stat = os.stat(self.contracts_dir)
+        source_mtime = self.get_source_modification_time()
+        no_cached = self._cached_compiled_contracts_mtime is None
 
-        if self._cached_compiled_contracts_stat != source_stat:
-            self._cached_compiled_contracts_stat = source_stat
+        if no_cached or self._cached_compiled_contracts_mtime < source_mtime:
```

Comment:
Maybe wrap this up into it's own helper method?  Something like `self.contract_sources_have_changed()` or whatever name you think is appropriate.